### ISSUE TITLE
Incorrectly reports a variable named `length` as an expensive use of `length()`

### ISSUE THREAD

**Issue by queer** [rrrene/credo#824]
The closest I've found to this bug is #720.

### Environment

* Credo version (`mix credo -v`): 1.5.1
* Erlang/Elixir version (`elixir -v`): Elixir 1.11.2 (compiled with Erlang/OTP 23)
* Operating system: Arch Linux

### What were you trying to do?

Consider this toy module:
```Elixir
defmodule Foo do
  def credo_you_are_wrong do
    length = 0
    if length == 0 do
      IO.puts "credo pls"
    end
  end
end
```
`length` is a variable here, not a list / not being used as a function, and is being compared to `0` in a way that makes sense to do so. credo reports this as an expensive use of `length()` instead.

### Expected outcome

credo understands that this is a variable and not a function call.

### Actual outcome

```
  Warnings - please take a look
┃ 
┃ [W] ↗ length == 0 is expensive. Prefer Enum.empty?/1 or list == []
┃       lib/foo.ex:4:8 #(Foo.credo_you_are_wrong)
```

**Comment by rrrene** [comment]
@queer Clearly a bug! Thx for reporting :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍
